### ISSUE TITLE
Files ignored by GIT aren't being linted

### ISSUE THREAD

**Issue by 123aswin123** [steelbrain/linter#1365]
Lint is not being applied to files ignored by VCS.

I did toggle the "Ignore VCS ignored " checkbox in the package settings. It doesn't seem to do anything.

I'm on OSX 10.11.6 , Atom version is 1.15.0 , Linter version is 2.1.0

**Comment by Arcanemagus** [comment]
Can you post the output of the debug command?

**Comment by steelbrain** [comment]
Hi @123aswin123 

After some further digging, I found out that we have no removed the option from linter that you're referring to, linter VCS ignoring is now managed with the setting `core.excludeVcsIgnoredPaths`, please change it's value to have the desired effect

**Comment by manuelbieh** [comment]
Same problem here. Was about to open an issue before finding this one here. My linter debug output:

```
Platform: win32
Atom Version: 1.15.0
Linter Version: 2.1.0
All Linter Providers: 
  - ESLint
  - sass-lint
```

Tried to enable and "re-disable" the "Ignore V C S ignored paths" option without any effect. Once I remove the paths from `.gitignore` files are linted again.

**Comment by manuelbieh** [comment]
Ah, I've just seen your `core.excludeVcsIgnoredPaths` explanation. Can't you re-add the feature then? I'd really like to have my `.gitignore` files and folders ignored by Atom itself (fuzzy finder etc.) but not by linter 😕

**Comment by cmseguin** [comment]
@steelbrain I agree with @manuelbieh. Why removing that feature? I have some config files that are ignored in my vcs, but I don't want to include vcs paths in core because I don't want the fuzzy search to be affected by a huge node_modules folder or something.

**Comment by steelbrain** [comment]
I must be misunderstanding something. You don't want node modules dir in fuzzy finder but you want to lint it?


Sent from my Samsung Galaxy smartphone.
-------- Original message --------From: Charles Mathieu Seguin <notifications@github.com> Date: 3/24/17  1:49 AM  (GMT+05:00) To: steelbrain/linter <linter@noreply.github.com> Cc: Steel Brain <i@steelbrain.me>, Mention <mention@noreply.github.com> Subject: Re: [steelbrain/linter] Files ignored by GIT aren't being linted (#1365) 
@steelbrain I agree with @manuelbieh. Why removing that feature? I have some config files that are ignored in my vcs, but I don't want to include vcs paths in core because I don't want the fuzzy search to be affected by a huge node_modules folder or something.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/steelbrain/linter","title":"steelbrain/linter","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/steelbrain/linter"}},"updates":{"snippets":[{"icon":"PERSON","message":"@cmseguin in #1365: @steelbrain I agree with @manuelbieh. Why removing that feature? I have some config files that are ignored in my vcs, but I don't want to include vcs paths in core because I don't want the fuzzy search to be affected by a huge node_modules folder or something. "}],"action":{"name":"View Issue","url":"https://github.com/steelbrain/linter/issues/1365#issuecomment-288855515"}}}

**Comment by manuelbieh** [comment]
Yes but to be honest, my setup might be a bit special. I have several projects in the company I'm currently working for where I have a structure like this:
```
- src/
- src/App
- src/Company
```
`src/Company` is in `.gitignore` and contains local versions of our company's node packages. We use symlinks to link to them from `node_modules` (e.g. `node_modules/my-package` -> `../src/Company/my-package`. `my-package` itself is a git repository (but not a git-submodule because ... submodules in git suck :wink: ) and it even contains its own `.eslintrc.json`.

Contrary to what I said earlier I don't have them in my fuzzy finder (which I rarely use anyway) but they got linted until recently.

Long story short: I deactivated `core.excludeVcsIgnoredPaths` and got my linting back. And I'm fine with it for the moment. But having an option to override the global `core.excludeVcsIgnoredPaths` would be pretty nice especially if the vcs ignored path even explicitly contains a `.eslintrc.json` file.

**Comment by cmseguin** [comment]
For me personally, at work we have a local config which is ignored in `.gitignore` (config-local.js)
and it's a node project so it also has a node_modules folder in `.gitignore`.
But I want to lint the config... If I turn off `core.excludeVcsIgnoredPaths`, it will lint my config, but slow down my fuzzy finder because of the node_modules.

Anyway, I did find a quick and dirty solution to my problem, but I was just wondering why removing an existing feature... I am not asking you to put it back, but it would have been nice to still have that feature

**Comment by Arcanemagus** [comment]
@cmseguin It was removed because we used to get issues filed from people confused as to why files ignored by the core setting were still getting linted. You can't please everyone it seems 😛

It's simpler for this to just follow the core setting, in the same manner that the `fuzzy-finder` package does for example.

**Comment by albertbuchard** [comment]
It took me 5 hours to find this issue - it should be made clearer.

**Comment by steelbrain** [comment]
@albertbuchard We now include this information in result of `Linter: Debug` command, you should invoke it and observe the output when experiencing something out of ordinary

**Comment by mqklin** [comment]
I need this feature as well. Added `excludeVcsIgnoredPaths: true` to `Atom/Config` and it works now.
